pkg/cmd/init/exec.go

```diff
@@ -192,12 +195,40 @@ func (o *Options) run() error {
 		}
 	}
 
-	fmt.Printf("The multicluster hub control plane has been initialized successfully!\n\n"+
-		"You can now register cluster(s) to the hub control plane. Log onto those cluster(s) and run the following command:\n\n"+
-		"    %s --cluster-name <cluster_name>\n\n"+
-		"Replace <cluster_name> with a cluster name of your choice. For example, cluster1.\n\n",
-		cmd,
-	)
+	if o.jsonOutput {
+		err := WriteJsonOutput(os.Stdout, token, restConfig.Host)
+		if err != nil {
+			return err
+		}
+	} else {
+		fmt.Printf("The multicluster hub control plane has been initialized successfully!\n\n"+
+			"You can now register cluster(s) to the hub control plane. Log onto those cluster(s) and run the following command:\n\n"+
+			"    %s --cluster-name <cluster_name>\n\n"+
+			"Replace <cluster_name> with a cluster name of your choice. For example, cluster1.\n\n",
+			cmd,
+		)
+	}
 
 	return apply.WriteOutput(o.outputFile, output)
 }
+
+func WriteJsonOutput(w io.Writer, token, host string) error {
+	val := HubValue{
+		HubToken:     token,
+		HubApiserver: host,
+	}
+	b, err := json.Marshal(val)
+	if err != nil {
+		return err
+	}
+	var out bytes.Buffer
+	err = json.Indent(&out, b, "", "")
```

Comment:
Using indent string of 2 spaces makes output little nicer:
 
    json.Indent(&out, b, "", "  ")